addons/product/tests/test_variants.py

```diff
@@ -909,19 +909,103 @@ def test_name_search_dynamic_attributes(self):
         template = self.env['product.template'].create({
             'name': 'cimanyd'
         })
-        self.assertEqual(len(template.product_variant_ids), 1)
 
         template.write({
             'attribute_line_ids': [(0, False, {
                 'attribute_id': dynamic_attr.id,
                 'value_ids': [(4, dynamic_attr.value_ids[0].id, False)],
             })]
         })
-        self.assertEqual(len(template.product_variant_ids), 0)
```

Comment:
I believe this assertion was necessary to ensure the `_name_search` override in the case of no variant was tested.
If you remove this, the test becomes useless because if there is a variant the test is not covering what it should.

You must adapt the setup of the test so that it stays in a situation where there is no variant generated in dynamic mode.

Reply:
Do you have an idea about how to do that? I looked at it but I don't see why adding an attribute line with a single value would change the length of `product_variant_ids` to 0. I think it worked before my changes because the variant was archived.

Reply:
I think the test should just create the product template by giving it immediately the correct `attribute_line_ids` instead of doing it as a second step. This way it should hopefully not create a variant in the first place.

Reply:
Indeed, thanks!